cmake/gtest/gtest.cmake

```diff
@@ -41,3 +41,9 @@ endif()
 
 # Add googletest, defines gtest and gtest_main targets
 add_subdirectory(${CMAKE_CURRENT_BINARY_DIR}/googletest-src ${CMAKE_CURRENT_BINARY_DIR}/googletest-build EXCLUDE_FROM_ALL)
+
+# Remove visibility.h from the compile flags for gtest because of poisoned exit()
+get_target_property(GTEST_COMPILE_FLAGS gtest COMPILE_OPTIONS)
```

Comment:
Longer term what we should do system wide is only apply our default flags, defines, includes, etc on a per target basis. Then in the gtest case we won't have this problem in the first place.

Reply:
That's exactly what I thought but I didn't want to dig over the whole build setup right now. We can still do this. Targets are definitely the way to go in contrast to including everything everywhere 😛. While blaming I saw that you are constantly cleaning up parts of the cmake configuration step by step. 👍